### ISSUE TITLE
Meshgen puts some faces in wrong mapblock mesh

### ISSUE THREAD

**Issue by numberZero** [minetest/minetest#6777]
Sometimes (on negative sides, at least for X and Z), faces on the mapblock edge belong not to the mapblock of their node, but to the next one. That causes funny half-visible nodes:
![screenshot_20171213_010230](https://user-images.githubusercontent.com/7352626/33910932-86390a12-dfa1-11e7-9e0e-7671d8b9c1d9.png)
and also requires several meshes to be updated on crack start at the edge, instead of only one.

**Comment by celeron55** [comment]
This works as intended - the faces have to belong to one, and only one mapblock mesh, otherwise they wouldn't be visible at all or would be rendered on top of each other if both mapblocks are visible.

**Comment by numberZero** [comment]
> the faces have to belong to one, and only one mapblock mesh

Of course. But why can’t they belong to the same mapblock as the node they belong to?

**Comment by paramat** [comment]
Because then the 2 nodes either side of a mapblock border would then both have a face on the border, so the border faces would be duplicated.

**Comment by numberZero** [comment]
> Because then the 2 nodes either side of a mapblock border would then both have a face on the border, so the border faces would be duplicated.

@paramat Of course, if you implement it wrongly.

**Comment by paramat** [comment]
Ok, reopening.

**Comment by paramat** [comment]
Possibly related #4925

**Comment by numberZero** [comment]
Certainly no. #4925 is about server-side issue, while this one is purely client-side.

**Comment by paramat** [comment]
But what's the issue here? The edge of the currently loaded mapblocks is usually distant and/or obscured by fog. I can't see a problem.

**Comment by paramat** [comment]
1 year bump. Any core dev support? (i'm neutral).
